src/components/EditCampaign.js

```diff
@@ -223,8 +215,33 @@ class EditCampaign extends React.Component {
         })
     }
 
+    async updateMaxAmount(that){
+        that.setState({showModal:true,
+            modalMessage: i18n.t('confirmMetamask'), errorIcon:'HourglassSplit',
+            modalButtonVariant: "gold", waitToClose: true})
+        if(!that.state.updateMaxAmount) return true;
+        try {
+            let result = await CAMPAIGNINSTANCE.methods.changeMaxAmount(web3.utils.toWei(this.state.maxAmount)).send({from:ACCOUNTS[0]}, () => {
+                that.setState({showModal:true, errorMessage: i18n.t('processingWait'),
+                modalMessage: i18n.t('updatingMaxAmount'), errorIcon:'HourglassSplit',
+                modalButtonVariant: "gold", waitToClose: true});
+            });
+            console.log(result);
+            return true;
+        } catch (error) {
+            that.setState({showModal : true,
+                errorMessage: i18n.t('blockChainTransactionFailed'),
+                modalMessage: i18n.t('checkMetamask'),
+                errorIcon:'XCircle', modalButtonMessage: i18n.t('closeBtn'),
+                modalButtonVariant: "#E63C36", waitToClose: false});
+            console.log("updating maxAmount transaction failed");
+            console.log(error);
```

Comment:
when transaction fails and code execution goes into "catch" block, this function returns "undefined". You should explicitly return "false" from the catch block